### PR TITLE
Handle different value types for constant op explicitly for TTNN backend

### LINKED ISSUE #1180: [TTNN Backend] Extract constant value with proper data type
TTNN backend is converting `ttir.constant` op which uses `mlir::ElementsAttr` to store the constant value. This constant value is extracted as
```
float fillValue = valueAttr.getElementType().isInteger()
                            ? static_cast<float>(valueAttr.getSplatValue<int>())
                            : valueAttr.getSplatValue<float>();
```

However, `getSplatValue<>()` requires exact data type for extracting the corresponding data. We cannot use `int` for all different integer types (like i1, i8, i16, i32...). Therefore, we need handle all integer data types explicitly. We can use `mlir::APFloat` to handle all floating point data types.

### PR DESCRIPTION
**Pull request by mmanzoorTT**
closes #1180

### CODE REVIEW

**Comment by sdjordjevicTT** [comment]
Can you please attach the relevant issue? Or if we don't have one, can you please create one?

**Comment by mmanzoorTT** [comment]
> Can you please attach the relevant issue? Or if we don't have one, can you please create one?

Created an issue and linked it.

File: test/ttmlir/Dialect/TTNN/simple_constant.mlir [hunk 1]
```diff
@@ -1,24 +1,60 @@
 // RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline %s | FileCheck %s
 #any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
 module attributes {} {
+  func.func @test_empty_int8() -> tensor<64x128xi8> {
+    %0 = "ttir.constant"() <{value = dense<0> : tensor<64x128xi8>}> : () -> tensor<64x128xi8>
+    // CHECK: %[[C:.*]] = "ttnn.empty"[[C:.*]]
```
[the remark this answers is not in the document]
Comment (mmanzoorTT) [reply test/ttmlir/Dialect/TTNN/simple_constant.mlir]: Thanks for the suggestion. I have added checks for fillValue and tensor shape.

**Review by sdjordjevicTT — APPROVED** [review]
It looks good from my side, thanks!

**Review by LPanosTT — APPROVED** [review]
Looks great 🔥

**Pull request merged**